src/main/java/com/dev/cinema/validation/FieldsValueMatchValidator.java

```diff
@@ -0,0 +1,25 @@
+package com.dev.cinema.validation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class FieldsValueMatchValidator
+        implements ConstraintValidator<FieldsValueMatch, Object> {
+
+    private String field;
+    private String fieldMatch;
+
+    @Override
+    public void initialize(FieldsValueMatch constraintAnnotation) {
+        this.field = constraintAnnotation.field();
+        this.fieldMatch = constraintAnnotation.fieldMatch();
+    }
+
+    @Override
+    public boolean isValid(Object value, ConstraintValidatorContext context) {
+        Object fieldValue = new BeanWrapperImpl(value).getPropertyValue(field);
+        Object fieldMatchValue = new BeanWrapperImpl(value).getPropertyValue(fieldMatch);
+        return fieldValue != null ? fieldValue.equals(fieldMatchValue) : fieldMatchValue == null;
```

Comment:
you can use Objects.equals() here instead

Reply:
done